Source/WebCore/svg/SVGComponentTransferFunctionElement.h

```diff
@@ -107,4 +109,31 @@ class SVGComponentTransferFunctionElement : public SVGElement {
     Ref<SVGAnimatedNumber> m_offset { SVGAnimatedNumber::create(this) };
 };
 
+inline FEComponentTransfer::Channel SVGComponentTransferFunctionElement::channel() const
+{
+    using namespace WebCore::ElementNames::SVG;
+    using Channel = FEComponentTransfer::Channel;
+
+    switch (elementName()) {
+    case feFuncR: return Channel::Red;
+    case feFuncG: return Channel::Green;
+    case feFuncB: return Channel::Blue;
+    case feFuncA: return Channel::Alpha;
+    default:
+        ASSERT_NOT_REACHED();
+        return Channel::Red;
+    }
+}
+
```

Comment:
Can not this be a pure virtual function in `SVGComponentTransferFunctionElement` and be overridden by the superclasses: `SVGFEFuncAElement`, `SVGFEFuncBElement`, etc?

Reply:
Yeah, I can do that. I guess it's more straightforward than computing it based on the element name.